### PR TITLE
[Codegen] Extract visit function to parsers/utils

### PR DESCRIPTION
**Pull request by AntoineDoubovetzky**
## Summary

This PR is a task from https://github.com/facebook/react-native/issues/34872:
> Extract the visit function in a shared function in the parsers/utils.js folder. Use the new function in the Flow and TypeScript parsers.

## Changelog

[Internal] [Changed] - Extract the visit function in a shared function in the parsers/utils.js

## Test Plan

I tested using jest, flow and eslint commands.

### CODE REVIEW

File: packages/react-native-codegen/src/parsers/__tests__/utils-test.js [hunk 1]
```diff
@@ -115,3 +116,107 @@ describe('createParserErrorCapturer', () => {
     });
   });
 });
+
+describe('visit', () => {
```
Comment (AntoineDoubovetzky) [inline packages/react-native-codegen/src/parsers/__tests__/utils-test.js:120]: I tried to test each branch, jest gives me 100% coverage but please tell me if you see a way to improve these tests.

Comment (cipolleschi) [reply packages/react-native-codegen/src/parsers/__tests__/utils-test.js:120]: They looks good to me!

**Review by cipolleschi — APPROVED** [review]
Thank you so much for the effort! It is amazing that we are also testing all this code! 🤩

**Comment by AntoineDoubovetzky** [comment]
I rebased the branch because of conflicts

**Comment by AntoineDoubovetzky** [comment]
@cipolleschi Do you need me to update something on this PR or is everything ok ?

**Comment by cipolleschi** [comment]
Thank you! No, nothing else from your side. I'll try to land this between today and tomorrow! 👍

**Pull request closed without merging**